CRM/Utils/System/Base.php

```diff
@@ -705,7 +705,7 @@ public function getTimeZoneOffset() {
 
       $timeZoneOffset = sprintf("%02d:%02d", $tz / 3600, abs(($tz / 60) % 60));
 
-      if ($timeZoneOffset > 0) {
+      if ($tz >= 0) { // $tz is a number, not a string, so it is better to test it.  00:00 needs a plus sign.
```

Comment:
Good thinking – when I was originally looking at the issue, I hit the challenge of a time zone of 00:00 giving an error which is why I changed the code.  Should we not be setting the MySQL time even if $tz is zero, since the server may have a different time zone?

 

 

From: Eileen McNaughton [mailto:notifications@github.com] 
Sent: 20 May 2016 03:46
To: civicrm/civicrm-core <civicrm-core@noreply.github.com>
Cc: Stephen Palmstrom <stephen.palmstrom@outlook.com>; Mention <mention@noreply.github.com>
Subject: Re: [civicrm/civicrm-core] CRM-15656 - synchronize PHP and MySQL time zones in Joomla (#8400)

 

In CRM/Utils/System/Base.php <https://github.com/civicrm/civicrm-core/pull/8400#discussion_r63984863> :

> @@ -705,7 +705,7 @@ public function getTimeZoneOffset() {
>  
>        $timeZoneOffset = sprintf("%02d:%02d", $tz / 3600, abs(($tz / 60) % 60));
>  
> -      if ($timeZoneOffset > 0) {
> +      if ($tz >= 0) { // $tz is a number, not a string, so it is better to test it.  00:00 needs a plus sign.

But why hasn't it already returned at this 

if (empty($tz)) {
return FALSE;
}

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub <https://github.com/civicrm/civicrm-core/pull/8400/files/03b8ad9e6c4d8435d444fba3b55496d126666a22#r63984863>   <https://github.com/notifications/beacon/AJkodwu1gf1vCdM84PtC42uwHWBq9Og7ks5qDSCBgaJpZM4Ihe5w.gif>